src/seedu/addressbook/data/tag/Tagging.java

```diff
@@ -0,0 +1,49 @@
+package seedu.addressbook.data.tag;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+import seedu.addressbook.data.person.Person;
+
+public class Tagging {
+	
+	private static final String ADD_REGEX = "+";
+	private static final String REMOVE_REGEX = "-";
+	
+	private final Person person;
+	private final Tag tag;
+	private boolean isRemoved; //if isRemoved is false, means Tag is added
+	
+	private static ArrayList<Tagging> tagLogList = new ArrayList<Tagging>();
```

Comment:
there is a difference whether we place this ArrayList here or in AddressBook. That's what the two diagrams are describing.